test/run_test.py

```diff
@@ -52,6 +52,14 @@
 
     HAVE_TEST_SELECTION_TOOLS = True
 except ImportError as e:
+
+    class ShardedTest(NamedTuple):
```

Comment:
This copy/paste seems prone to divergence...how about extracting ShardedTest to a separate module instead which doesn't need boto3?

Reply:
Modified `run_test.py` so it does not need `ShardedTest` at all